src/monster.cpp

```diff
@@ -2139,6 +2153,37 @@ void monster::die( Creature *nkiller )
     }
 }
 
+bool monster::mod_mech_power( int amt )
+{
+    if( is_hallucination() || !has_flag( MF_RIDEABLE_MECH ) || !battery_item ) {
+        return false;
+    }
+    amt = -amt;
+    battery_item->ammo_consume( amt, pos() );
+    if( battery_item->ammo_remaining() <= 0 ) {
+        return false;
+    }
+    return true;
+}
+
+bool monster::check_mech_powered()
+{
+    if( is_hallucination() || !has_flag( MF_RIDEABLE_MECH ) || !battery_item ) {
+        return false;
+    }
+    if( battery_item->ammo_remaining() <= 0 ) {
+        return false;
+    }
+    itype_id mech_bat = itype_id( type->mech_battery );
```

Comment:
```suggestion
    const itype_id &mech_bat = type->mech_battery;
```
The (explicit!) copy is not needed.